### PR TITLE
[SPARK-14667] Remove HashShuffleManager

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

The sort shuffle manager has been the default since Spark 1.2. It is time to remove the old hash shuffle manager.
## How was this patch tested?

Removed some tests related to the old manager.

### CODE REVIEW

**Comment by rxin** [comment]
I will leave this open for a few days to give other people a chance to comment.

**Comment by witgo** [comment]
I think the following should be deleted
1.  [FileShuffleBlockResolver.scala](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala)
2. [getHashBasedShuffleBlockData in ExternalShuffleBlockResolver](https://github.com/apache/spark/blob/master/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java#L257)

**Comment by rxin** [comment]
Thanks - updated.

File: common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolverSuite.java [hunk 1]
```diff
@@ -118,27 +112,6 @@ public void testSortShuffleBlocks() throws IOException {
   }
 
   @Test
-  public void testHashShuffleBlocks() throws IOException {
-    ExternalShuffleBlockResolver resolver = new ExternalShuffleBlockResolver(conf, null);
```
Comment (JoshRosen) [inline common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolverSuite.java]: If I'm not mistaken, I think that this code path is exercised by the external shuffle service, so a Spark 2.x shuffle service would be unable to serve shuffle files generated by Spark 1.x when using hash shuffle.

Comment (rxin) [inline common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolverSuite.java]: Yup I think that's fine and an explicit decision?

**Comment by rxin** [comment]
Alright merging in master. Thanks.

**Pull request closed without merging**